### PR TITLE
Support boolean COO sparse matrix

### PR DESCRIPTION
**Pull request by asi1024**
Part of #7741.

### CODE REVIEW

File: cupyx/scipy/sparse/_coo.py [hunk 1]
```diff
@@ -414,7 +415,18 @@ def sum_duplicates(self):
             data = cupy.zeros(size, dtype=self.data.dtype)
             row = cupy.empty(size, dtype='i')
             col = cupy.empty(size, dtype='i')
-            if self.data.dtype.kind == 'f':
+            if self.data.dtype.kind == 'b':
+                cupy.ElementwiseKernel(
+                    'T src_data, int32 src_row, int32 src_col, int32 index',
+                    'raw T data, raw int32 row, raw int32 col',
+                    '''
+                    if (src_data) data[index] = true;
```
Comment (emcastillo) [inline cupyx/scipy/sparse/_coo.py:423]:
```suggestion
                    if (src_data) data[index] = true;
```
Guess this is also related to #7724 :)

**Review by emcastillo — APPROVED** [review]

**Comment by asi1024** [comment]
@emcastillo PTAL!

**Comment by emcastillo** [comment]
Test is unrelated

**Review by emcastillo — APPROVED** [review]

**Pull request merged**